src/Analysis/Ast/Impl/Extensions/PythonModuleExtensions.cs

```diff
@@ -26,5 +28,23 @@ internal static PythonAst GetAst(this IPythonModule module)
             => ((IAstNodeContainer)module).GetAstNode<T>(o);
         internal static void AddAstNode(this IPythonModule module, object o, Node n)
             => ((IAstNodeContainer)module).AddAstNode(o, n);
+
+        /// <summary>
+        /// Returns the string line corresponding to the given location
+        /// </summary>
+        /// <param name="line">The line number</param>
+        internal static string GetLine(this PythonModule module, int line) {
+            SourceLocation source = new SourceLocation(line, 1);
+            string content = module.Content;
+            var start = GetAst(module).LocationToIndex(source);
+            var length = 0;
+
+            if (!content.IsNullOrEmpty()) {
+                while(start + length < content.Length && content[start + length] != '\n' && content[start + length] != '\r') {
```

Comment:
I'd take a look at GetLineIndent; the loop style used there is cleaner than having all of these `start + length` all over the place.